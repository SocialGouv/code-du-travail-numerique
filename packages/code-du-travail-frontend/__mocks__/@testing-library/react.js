/* eslint-disable import/export */
import { theme } from "@socialgouv/cdtn-ui";
import { render } from "@testing-library/react";
import PropTypes from "prop-types";
import React from "react";
import { ThemeProvider } from "styled-components";

const AllTheProviders = ({ children }) => {
  return (
    <ThemeProvider theme={theme.colors}>
      <>{children}</>
    </ThemeProvider>
  );
};

AllTheProviders.propTypes = {
  children: PropTypes.node.isRequired,
};

const customRender = (ui, options) =>
  render(ui, { wrapper: AllTheProviders, ...options });

// re-export everything
export * from "@testing-library/react";

// override render method
export { customRender as render };
