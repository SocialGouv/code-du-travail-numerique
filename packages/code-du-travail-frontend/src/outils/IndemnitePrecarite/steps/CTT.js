import { Toast } from "@socialgouv/cdtn-ui";
import React from "react";
import { Field } from "react-final-form";

import { YesNoQuestion } from "../../common/YesNoQuestion";

function validate(values) {
  const errors = {};
  if (values.cttFormation === true) {
    errors.cttFormation =
      "Ce type de contrat ne permet pas au salarié d’avoir droit à une prime de précarité.";
  }
  if (values.ruptureContratFauteGrave === true) {
    errors.ruptureContratFauteGrave =
      "Lorsque le contrat de travail temporaire (contrat d'intérim) est rompu de manière anticipée à l’initiative du salarié, pour faute grave du salarié ou en cas de force majeure, le salarié n’a pas le droit à une prime de précarité.";
  }
  if (values.propositionCDIFinContrat === true) {
    errors.propositionCDIFinContrat =
      "Le salarié en contrat de travail temporaire (contrat d’intérim) qui est immédiatement embauché en CDI au sein de l’entreprise dans laquelle il effectuait sa mission n’a pas le droit à une prime de précarité.";
  }
  if (values.refusSouplesse === true) {
    errors.refusSouplesse =
      "Le salarié en contrat d’intérim qui refuse la mise en œuvre de la souplesse prévue dans son contrat n’a pas le droit à une prime de précarité.";
  }
  return errors;
}

function StepCTT() {
  return (
    <>
      <YesNoQuestion
        label="S’agit-il d’un contrat de mission-formation&nbsp;?"
        name="cttFormation"
        data-testid="cttFormation"
        autoFocus
      />
      <Field name="cttFormation">
        {({ input }) =>
          input.value === false ? (
            <Toast>
              Attention&nbsp;: s’il s’agit d’un contrat de travail temporaire
              saisonnier ou d’usage, un accord d’entreprise ou d’établissement
              peut dispenser l’entreprise de travail temporaire (l’entreprise
              d’intérim) de verser la prime de précarité.
            </Toast>
          ) : null
        }
      </Field>
      <YesNoQuestion
        data-testid="ruptureContratFauteGrave"
        name="ruptureContratFauteGrave"
        label="Le contrat d'intérim a-t-il été rompu avant la fin prévue pour une des raisons suivantes&nbsp;: la propre initiative du salarié, la faute grave du salarié, cas de force majeure&nbsp;?"
      />
      <YesNoQuestion
        data-testid="propositionCDIFinContrat"
        name="propositionCDIFinContrat"
        label="À la fin du contrat d'intérim, le salarié a-t-il été immédiatement embauché en CDI au sein de l'entreprise dans laquelle il effectuait sa mission&nbsp;?"
      />
      <YesNoQuestion
        data-testid="refusSouplesse"
        name="refusSouplesse"
        label="Le salarié a-t-il refusé la mise en œuvre de la souplesse prévue dans le contrat d’intérim&nbsp;?"
      />
    </>
  );
}

StepCTT.validate = validate;

export { StepCTT };
