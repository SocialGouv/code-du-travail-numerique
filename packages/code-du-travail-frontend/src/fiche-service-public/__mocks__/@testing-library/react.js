import React from "react";
import PropTypes from "prop-types";
import { render } from "@testing-library/react";
import { ThemeProvider } from "styled-components";
import { theme } from "@socialgouv/cdtn-ui";

const AllTheProviders = ({ children }) => {
  return (
    <ThemeProvider theme={theme.colors}>
      <>{children}</>
    </ThemeProvider>
  );
};

AllTheProviders.propTypes = {
  children: PropTypes.node.isRequired,
};

const customRender = (ui, options) =>
  render(ui, { wrapper: AllTheProviders, ...options });

// re-export everything
export * from "@testing-library/react";

// override render method
export { customRender as render };
