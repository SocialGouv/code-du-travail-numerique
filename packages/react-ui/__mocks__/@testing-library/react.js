import { render } from "@testing-library/react";
import PropTypes from "prop-types";
import React from "react";
import { ThemeProvider } from "styled-components";

import { colors } from "../../src/theme";

const AllTheProviders = ({ children }) => {
  return (
    <ThemeProvider theme={colors}>
      <>{children}</>
    </ThemeProvider>
  );
};

AllTheProviders.propTypes = {
  children: PropTypes.node.isRequired,
};

const customRender = (ui, options) =>
  render(ui, { wrapper: AllTheProviders, ...options });

// re-export everything
export * from "@testing-library/react";

// override render method
export { customRender as render };
