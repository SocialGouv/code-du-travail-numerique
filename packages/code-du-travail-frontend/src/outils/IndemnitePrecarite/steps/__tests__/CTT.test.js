import { fireEvent, render } from "@testing-library/react";
import React from "react";
import { Form } from "react-final-form";

import { StepCTT } from "../CTT";

function renderForm() {
  return render(
    <Form validate={StepCTT.validate} onSubmit={jest.fn()}>
      {({ handleSubmit, form }) => (
        <form onSubmit={handleSubmit}>
          <StepCTT form={form} />
          <button data-testid="nextBt">suivant</button>
        </form>
      )}
    </Form>
  );
}

const testQuestions = [
  ["ruptureContratFauteGrave", true],
  ["propositionCDIFinContrat", true],
  ["refusSouplesse", true],
];

describe("<StepCTT />", () => {
  it("should render", () => {
    const { container } = renderForm();
    expect(container).toMatchSnapshot();
  });

  test.each(testQuestions)(
    "should render error when %s is %s",
    (key, value) => {
      const { getByText, getByTestId } = renderForm();
      const radio = getByTestId(key).querySelector(`input[value=${value}]`);
      fireEvent.click(radio);
      // validate form
      const validateButton = getByTestId("nextBt");
      validateButton.click();
      expect(getByText(/pas le droit Ã  une prime/i)).toBeTruthy();
    }
  );
});
