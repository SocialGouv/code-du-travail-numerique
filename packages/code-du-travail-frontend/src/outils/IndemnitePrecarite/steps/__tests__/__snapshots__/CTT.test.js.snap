// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<StepCTT /> should render 1`] = `
.c1 {
  color: #3e486e;
  font-size: 1.6rem;
  font-weight: 400;
}

.c5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  font-size: 1.6rem;
  cursor: pointer;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
}

.c4 {
  position: relative;
  display: inline-block;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  width: calc(1.6rem * 1.25);
  height: calc(1.6rem * 1.25);
  margin: calc(1.6rem / 5) calc(2 * 1.6rem / 3) 0 0;
  padding: 0;
  line-height: inherit;
  background: #fff;
  border: 1px solid #bbcadf;
  border-radius: 50%;
  box-shadow: none;
  cursor: pointer;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
}

.c4:before {
  position: absolute;
  top: 50%;
  left: 50%;
  display: none;
  width: calc(1.6rem / 1.5);
  height: calc(1.6rem / 1.5);
  background-color: #f66663;
  border-radius: 50%;
  -webkit-transform: translate(-50%,-50%);
  -ms-transform: translate(-50%,-50%);
  transform: translate(-50%,-50%);
  content: "";
}

.c4:checked {
  background-color: #fff;
  border-color: #f66663;
}

.c4:checked:before {
  display: block;
}

.c0 {
  display: inline-block;
  margin-top: 2rem;
  margin-bottom: 1rem;
  font-size: 1.8rem;
  font-weight: 600;
  cursor: pointer;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  margin-bottom: 2rem;
}

@media (max-width:600px) {
  .c0 {
    font-size: 1.6rem;
  }
}

<div>
  <form>
    <label
      class="c0"
    >
      S’agit-il d’un contrat de mission-formation ?
      <span
        class="c1"
      >
         (obligatoire)
      </span>
    </label>
    <div
      class="c2"
      data-testid="cttFormation"
    >
      <div
        class="c3"
      >
        <input
          class="c4"
          id="cttFormation-oui"
          name="cttFormation"
          type="radio"
          value="true"
        />
        <label
          class="c5"
          for="cttFormation-oui"
        >
          Oui
        </label>
      </div>
      <div
        class="c3"
      >
        <input
          class="c4"
          id="cttFormation-non"
          name="cttFormation"
          type="radio"
          value="false"
        />
        <label
          class="c5"
          for="cttFormation-non"
        >
          Non
        </label>
      </div>
    </div>
    <label
      class="c0"
    >
      Le contrat d'intérim a-t-il été rompu avant la fin prévue pour une des raisons suivantes : la propre initiative du salarié, la faute grave du salarié, cas de force majeure ?
      <span
        class="c1"
      >
         (obligatoire)
      </span>
    </label>
    <div
      class="c2"
      data-testid="ruptureContratFauteGrave"
    >
      <div
        class="c3"
      >
        <input
          class="c4"
          id="ruptureContratFauteGrave-oui"
          name="ruptureContratFauteGrave"
          type="radio"
          value="true"
        />
        <label
          class="c5"
          for="ruptureContratFauteGrave-oui"
        >
          Oui
        </label>
      </div>
      <div
        class="c3"
      >
        <input
          class="c4"
          id="ruptureContratFauteGrave-non"
          name="ruptureContratFauteGrave"
          type="radio"
          value="false"
        />
        <label
          class="c5"
          for="ruptureContratFauteGrave-non"
        >
          Non
        </label>
      </div>
    </div>
    <label
      class="c0"
    >
      À la fin du contrat d'intérim, le salarié a-t-il été immédiatement embauché en CDI au sein de l'entreprise dans laquelle il effectuait sa mission ?
      <span
        class="c1"
      >
         (obligatoire)
      </span>
    </label>
    <div
      class="c2"
      data-testid="propositionCDIFinContrat"
    >
      <div
        class="c3"
      >
        <input
          class="c4"
          id="propositionCDIFinContrat-oui"
          name="propositionCDIFinContrat"
          type="radio"
          value="true"
        />
        <label
          class="c5"
          for="propositionCDIFinContrat-oui"
        >
          Oui
        </label>
      </div>
      <div
        class="c3"
      >
        <input
          class="c4"
          id="propositionCDIFinContrat-non"
          name="propositionCDIFinContrat"
          type="radio"
          value="false"
        />
        <label
          class="c5"
          for="propositionCDIFinContrat-non"
        >
          Non
        </label>
      </div>
    </div>
    <label
      class="c0"
    >
      Le salarié a-t-il refusé la mise en œuvre de la souplesse prévue dans le contrat d’intérim ?
      <span
        class="c1"
      >
         (obligatoire)
      </span>
    </label>
    <div
      class="c2"
      data-testid="refusSouplesse"
    >
      <div
        class="c3"
      >
        <input
          class="c4"
          id="refusSouplesse-oui"
          name="refusSouplesse"
          type="radio"
          value="true"
        />
        <label
          class="c5"
          for="refusSouplesse-oui"
        >
          Oui
        </label>
      </div>
      <div
        class="c3"
      >
        <input
          class="c4"
          id="refusSouplesse-non"
          name="refusSouplesse"
          type="radio"
          value="false"
        />
        <label
          class="c5"
          for="refusSouplesse-non"
        >
          Non
        </label>
      </div>
    </div>
    <button
      data-testid="nextBt"
    >
      suivant
    </button>
  </form>
</div>
`;
